### DISCUSSION TITLE
Fail2ban 1.0.2 with newsyslog signal for rotate log

### DISCUSSION THREAD

**Discussion by ysha0** [fail2ban/fail2ban#3501] [General]
Hi, using Fail2ban 1.0.2 on FreeBSD, trying configure for rotate log:

/etc/newsyslog.conf
`/var/log/fail2ban.log                   600  5     100  *     JC /var/run/fail2ban/fail2ban.pid`

After rotating the log by newsyslog, Fail2ban stops writing the log to the new rotated empty log file /var/log/fail2ban.log until restart Fail2ban
`Apr 24 05:00:00 srv newsyslog[59281]: logfile turned over due to size>100K`

If the program does not understand that there is a rotation of the log, then the standard method is `JC /var/run/fail2ban/fail2ban.pid`. But it doesn't work.

Now I'm testing another option:
/etc/newsyslog.conf
`/var/log/fail2ban.log                   600  5     100  *     JC /var/run/fail2ban/fail2ban.pid 30`

Discussion of a similar problem on FreeBSD https://forums.freebsd.org/threads/fail2ban-stops-working-when-logs-are-rotated.58737

How right to send signal for running Fail2Ban process about log was rotated?
It is also unclear whether Fail2ban only stops writing to the log after rotation or ceases to function at all after log rotation?

**Comment by sebres** [comment]
Invoke `fail2ban-client` with `flushlogs`, see for example
https://github.com/fail2ban/fail2ban/blob/2b98f461bb93c6a1a1cec1481fa94b7a84342977/files/fail2ban-logrotate#L8-L14

**Reply by ysha0** [reply]
Thanks to your information and https://web.archive.org/web/20190612093808/http://fuzzix.org/post/33
Now it works on FreeBSD as well:

/etc/newsyslog.conf
`/var/log/fail2ban.log                   600  5     100  *     JR /usr/local/bin/f2b_flogs.sh`

/usr/local/bin/f2b_flogs.sh
`#!/bin/sh
/usr/local/bin/fail2ban-client flushlogs >/dev/null`

/var/log/fail2ban.log after rotate:
`Apr 24 21:00:00 newsyslog[96068]: logfile turned over due to size>100K`
`2023-04-24 21:00:00,079 fail2ban.server [94820]: INFO rollover performed on /var/log/fail2ban.log`

**Reply by Munktells** [reply]
> Invoke `fail2ban-client` with `flushlogs`, see for example

How about newsyslog?
https://man.freebsd.org/cgi/man.cgi?newsyslog

Is logrotate the only one that can rotate fail2ban logs?

https://forums.freebsd.org/threads/cant-rotate-fail2ban-log-with-newsyslog-in-freebsd-15.102315/
